### PR TITLE
refactor: introduce page

### PR DESCRIPTION
**Pull request by ashleygwilliams**
continuing work from #135

### CODE REVIEW

**Review by SaraVieira — APPROVED** [review]

File: src/site/artifacts/page.rs [hunk 1]
```diff
@@ -42,11 +41,6 @@ pub fn build(config: &Config) -> Result<()> {
...
         html.extend(cargo_dist::build_table(manifest, config));
-    };
-
-    let doc = layout::build(config, html!(<div>{html}</div>), false)?;
-    let html_path = format!("{}/artifacts.html", &config.dist_dir);
-    let asset = axoasset::local::LocalAsset::new(&html_path, doc.into());
-    axoasset::local::LocalAsset::write(&asset, &config.dist_dir)?;
-    Ok(())
+    }
+    Ok(html!(<div>{html}</div>).to_string())
```
Comment (ashleygwilliams) [inline src/site/artifacts/page.rs:45]: @SaraVieira should i be worried about adding this extra div here? it was the only thing i could do to get this going...

Comment (SaraVieira) [reply src/site/artifacts/page.rs:45]: Unless it breaks the layout it's all good

Comment (ashleygwilliams) [reply src/site/artifacts/page.rs:45]: ok it doesn't seem to

File: src/commands/build.rs [hunk 2]
```diff
@@ -20,7 +20,7 @@ impl Build {
         Message::new(MessageType::Info, "Running build...").print();
         tracing::info!("Running build...");
         let config = Config::build(&self.config_path)?;
-        Site::write(&config)?;
+        Site::build(&config)?.write(&config)?;
```
Comment (SaraVieira) [inline src/commands/build.rs:23]: nice

**Pull request merged**